modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -845,16 +889,22 @@ class StreamingOutput final: public cv::gimpl::GIslandExecutable::IOutput
         while (post_iter != m_postings[out_idx].end() && post_iter->ready == true)
         {
             Cmd cmd;
-            if (cv::util::holds_alternative<cv::GRunArg>(post_iter->data))
-            {
-                cmd = Cmd{cv::util::get<cv::GRunArg>(post_iter->data)};
-            }
-            else
+            switch (post_iter->data.index())
             {
-                GAPI_Assert(cv::util::holds_alternative<cv::gimpl::EndOfStream>(post_iter->data));
-                cmd = Cmd{Stop{}};
-                m_stops_sent++;
+                case Posting::V::index_of<cv::GRunArg>():
+                    cmd = Cmd{cv::util::get<cv::GRunArg>(post_iter->data)};
+                    break;
+                case Posting::V::index_of<cv::gimpl::Exception>():
+                    cmd = Cmd{cv::util::get<cv::gimpl::Exception>(post_iter->data)};
+                    break;
+                case Posting::V::index_of<cv::gimpl::EndOfStream>():
+                    cmd = Cmd{Stop{}};
+                    m_stops_sent++;
```

Comment:
I didn't get the idea, why it's needed?

Reply:
not sure, for statistics maybe - in case of multiple exception are allowed in pipeline without stopping it

If this case is not supported - then please ignore this comment

Reply:
Multiple exceptions aren't supported.

In this PR, I try not to allow exceptions to be thrown from not `main` thread.